elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/engine/StitchList.java

```diff
@@ -0,0 +1,156 @@
+     *
+     * @return {@code true} if the entity instances in this {@link StitchList} should be further hydrated because they
+     * have one or more relationship fields.
+     */
+    public boolean shouldStitch() {
+        return !getTodoList().isEmpty();
+    }
+
+    /**
+     * Enqueues an entity instance which will be further hydrated on one of its relationship fields later
+     *
+     * @param entityInstance  The entity instance to be hydrated
+     * @param fieldName  The relationship field to hydrate in the entity instance
+     * @param value  The foreign key between the entity instance and the field entity.
+     */
+    public void todo(Object entityInstance, String fieldName, Object value) {
+        getTodoList().add(new Todo(entityInstance, fieldName, value));
+    }
+
+    /**
+     * Sets all the relationship values of an requested entity.
+     * <p>
+     * Values associated with the existing key will be overwritten.
+     *
+     * @param relationshipType  The type of the relationship to set
+     * @param idToInstance  A map from relationship ID to the actual relationship instance with that ID
+     */
+    public void populateLookup(Class<?> relationshipType, Map<Object, Object> idToInstance) {
+        getObjectLookups().put(relationshipType, idToInstance);
+    }
+
+    /**
+     * Stitch all entity instances currently in this {@link StitchList} by setting their relationship fields whose
+     * values are determined by relationship ID's.
+     */
+    public void stitch() {
+        for (Todo todo : getTodoList()) {
+            Object entityInstance = todo.getEntityInstance();
+            String relationshipName = todo.getRelationshipName();
+            Object foreignKey = todo.getForeignKey();
+
+            Class<?> relationshipType = getEntityDictionary().getType(entityInstance, relationshipName);
+            Object relationshipValue = getObjectLookups().get(relationshipType).get(foreignKey);
+
+            getEntityDictionary().setValue(entityInstance, relationshipName, relationshipValue);
+        }
+    }
+
+    /**
+     * Returns a mapping from relationship name to an immutable list of foreign key ID objects.
+     * <p>
+     * For example, given the following {@code todoList}:
+     * <pre>
+     * {@code
+     *     [PlayerStats, country, 344]
+     *     [PlayerStats, country, 840]
+     *     [PlayerStats, country, 344]
+     *     [PlayerStats, player, 1]
+     *     [PlayerStats, player, 1]
+     *     [PlayerStats, player, 1]
+     * }
+     * </pre>
+     * this method returns a map of the following:
+     * <pre>
+     *     [
+     *         "country": [344, 840]
+     *         "player": [1]
+     *     ]
+     * </pre>
+     *
+     * @return a mapping from relationship name to an ordered list of relationship join ID's
+     */
+    public Map<String, List<Object>> getHydrationMapping() {
+        return getTodoList().stream()
```

Comment:
This also looks wrong.  If the todo list is managing multiple types, it is blindly throwing away the type information and just grouping on the relationship name.  What if two todos have the same relationship name but different types?